### ISSUE TITLE
When login to bo with version 2.11.x, front web page are not load correctly

### ISSUE THREAD

**Issue by cedrox** [wp-media/wp-rocket#713]
When activate, plugin crash the instance.

Concatenate static file function of wp rocket is not working propertly on version :
- 2.11.2
- 2.11.4
- 2.11.1 (trouble only header and footer)

Roll back to 2.10.12

**Comment by xbuzz** [comment]
Seeing this behavior as well. Static concatenation of styles fails to generate and save the concat stylesheet and isn’t inserted into the cached page. I do see a blank style sheet with an empty href in the link tag in the head.

**Comment by Keomoniroath** [comment]
Style.css is not loaded with the concat stylesheet. Print screen of the blank stylesheet : 

![image](https://user-images.githubusercontent.com/29626942/35109191-f1a405ea-fc75-11e7-8e31-0a36430ea956.png)

**Comment by Tabrisrp** [comment]
Please provide more feedback about the bug, especially how to accurately reproduce it, if you want to file a bug report.

If you're looking for support, please open a support ticket.

**Comment by xbuzz** [comment]
@Tabrisrp I did some digging. The issue I noticed was introduced in the `v2.11` release back in December. The specific issue I'm seeing appears to be related to the change where WP Rocket no longer relies on HTTP requests (kick ass change btw).

It appears that the value of `WP_SITEURL` (via `site_url()`) is being used to determine paths instead of `WP_HOME` (`home_url()`). Looking through the source this issue may exist in multiple places (missing styles, is just a bit more obvious).

So looking at the example config below WP Rocket is looking at

http://dev.example.com/wp/wp-content/themes/example/somestyles.css (incorrect)

instead of the correct 

http://dev.example.com/wp-content/themes/example/somestyles.css

Current config (broken in v2.11)

```php
// current config
define('WP_HOME', 'http://dev.example.com');
define('WP_SITEURL', 'http://dev.example.com/wp');
```

Modified (works) 

```php
// current config
define('WP_HOME', 'http://dev.example.com');
define('WP_SITEURL', 'http://dev.example.com');
```

So if WP core `WP_SITEURL` differs from the actual site then things break.

Happy to help you dig in further but it appears that `site_url()` is used instead of the correct `home_url()`.

Let me know if you need any additional details.

**Comment by xbuzz** [comment]
To clarify, the above `site_url()` vs `home_url()` was just a best guess. Moving WP core assets to the default install path and updating `WP_SITEURL` "fixed" things. I'd need a bit more time to further debug, but it does appear that this combined w/ install WP core in a subdirectory isn't playing nicely.

If anyone else seeing similar issues could provide a bit more detail w/ their WP installation to see if there are any other similarities.

**Comment by Tabrisrp** [comment]
`site_url()` is the path where WordPress is installed, and supposedly contains all WordPress files, plugins and themes. That is why it is used in place of `home_url()`.

A setup with a different `home_url()` and `site_url()` has been tested and works without issue. Did you make additional changes?

**Comment by Keomoniroath** [comment]
This is not the solution for us because we have the same url ( site_url() and home_url() ), they're generated dynamically. 
By the way we are on Azure, maybe it's linked to the bug ?

**Comment by Tabrisrp** [comment]
@Keomoniroath we do have a bug currently with windows servers #705 so it's very possible it's that for you

**Comment by Keomoniroath** [comment]
Sorry I should give more details, we are in a Linux container on Azure

**Comment by di5abled** [comment]
We also have this problem, and I think I was able to pin down where the problem happens. You are making a str_replace where no matches are found: https://github.com/wp-media/wp-rocket/blob/master/inc/functions/formatting.php#L267

This happens when you try to generate the absolut path to the file from an given url.

Here are some logs I collected. `$file` should be the absolute path to the asset.

```
$site_url: http://domain/wp/
$abspath: /path/public/wp/
$file: http://domain/wp_contents/plugins/woocommerce-multilingual/res/js/front-scripts.js

$site_url: http://domain/wp/
$abspath: /path/public/wp/
$file: http://domain/wp_contents/plugins/woocommerce-multilingual/res/js/cart_widget.js

$site_url: http://domain/wp/
$abspath: /path/public/wp/
$file: http://domain/wp_contents/plugins/woocommerce/assets/js/frontend/add-to-cart.js

$site_url: http://domain/wp/
$abspath: /path/public/wp/
$file: http://domain/wp_contents/plugins/woocommerce/assets/js/frontend/woocommerce.js

$site_url: http://domain/wp/
$abspath: /path/public/wp/
$file: http://domain/wp_contents/plugins/woocommerce/assets/js/frontend/cart-fragments.js
```

Our WordPress installation is located in the `/wp` subfolder, so our `WP_SITEURL` points to this subfolder, which we also see in the `$site_url` variable. Maybe consider using `WP_HOME` which only has the domain eg. `http://domain`

Our `CONTENT_DIR` ist the `/wp_contents` folder which is outside of the `/wp` subfolder.

Because of this you try to replace a `$site_url` which isn't used in the assets url, which means we don't get an absolute url and the origin assets url is still in the `$file` variable which can't be minified.

**Comment by Tabrisrp** [comment]
Ok so the issue is specific to installation where the wp-content directory is moved outside of the WordPress installation directory?

**Comment by di5abled** [comment]
seems like, yeah

**Comment by xbuzz** [comment]
This sounds like it should be fixed with #729. See also the related issue in that PR.

**Comment by Tabrisrp** [comment]
Should be fixed in 2.11.5

**Comment by cedrox** [comment]
We will try the new version. Thanks a lot :)
